react/src/components/QrReaderOverlay/QrReaderOverlayContainer.tsx

```diff
@@ -50,127 +44,163 @@ const boxDataToSuccessQrValue = (boxData: BoxDetailsQuery["box"]) => {
-              // same as the current base of the user.
-              // E.g. introduce another non-success kind for this case.
-              const resolvedQrValueWrapper = {
-                ...qrValueWrapper,
-                isLoading: false,
-                finalValue: boxDataToSuccessQrValue(data?.qrCode?.box),
-              } as IQrValueWrapper;
-              addQrValueWrapperToMap(resolvedQrValueWrapper);
+              notificationVar({
+                title: "QR Code",
+                type: "error",
+                message: "Error: The QR code could not be found",
+              });
             }
+          })
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error);
+            //   const resolvedQrValueWrapper = {
+            //     ...qrValueWrapper,
+            //     isLoading: false,
+            //     finalValue: { kind: "noBoxtributeQr" },
+            //   } as IQrValueWrapper;
+            //   return Promise.resolve(resolvedQrValueWrapper);
           });
-        // TODO: Handle Authorization / No Access To Box case
-        // .catch((error) => {
-        //   console.error(error);
-        //   const resolvedQrValueWrapper = {
-        //     ...qrValueWrapper,
-        //     isLoading: false,
-        //     finalValue: { kind: "noBoxtributeQr" },
-        //   } as IQrValueWrapper;
-        //   return Promise.resolve(resolvedQrValueWrapper);
-        // });
       }
     },
-    [addQrValueWrapperToMap, apolloClient]
+    [addQrValueWrapperToMap, apolloClient],
   );
 
   const handleSingleScan = useCallback(
     (result: string) => {
       const qrCode = extractQrCodeFromUrl(result);
       if (qrCode == null) {
-        console.error("Not a Boxtribute QR Code");
-        onScanningDone([{ kind: "noBoxtributeQr" }]);
+        notificationVar({
+          title: "Error",
+          type: "error",
+          message: "Not a Boxtribute QR Code",
+        });
+        onScanningDone([{ kind: QrResolverResultKind.NOT_BOXTRIBUTE_QR }]);
       } else {
         apolloClient
-          .query<
-            GetBoxLabelIdentifierForQrCodeQuery,
-            GetBoxLabelIdentifierForQrCodeQueryVariables
-          >({
-            query: GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
-            fetchPolicy: "no-cache",
-            variables: { qrCode },
-          })
-          .then(({ data }) => {
+          .query<GetBoxLabelIdentifierForQrCodeQuery, GetBoxLabelIdentifierForQrCodeQueryVariables>(
+            {
+              query: GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
+              fetchPolicy: "no-cache",
+              variables: { qrCode },
+            },
+          )
+          .then(({ data, errors }) => {
             const boxLabelIdentifier = data?.qrCode?.box?.labelIdentifier;
-            if (boxLabelIdentifier == null) {
+
+            if ((errors?.length || 0) > 0) {
```

Comment:
There needs to be more distinction of the error code. 
1. If the QR code is not assigned to a box, looking up the box fails, and the error formatter in `exceptions.py` sets the `BAD_USER_REQUEST` GraphQL error code (unfortunately this is only the behavior for production mode, not for development mode). This should be the `NOT_ASSIGNED_TO_BOX` result.
2. If the QR code is assigned to a box but the user is not authorized to access that box, a `FORBIDDEN` error is returned. This should be the `NOT_AUTHORIZED` result.

Reply:
@pylipp  Thanks for your suggestion, I handled the error in the FE, and also wrote on more cases for unhandled exceptions on BE (if any)